File: app/src/main/java/com/wangzhen/adapter/sample/holder/EmptyHolder.java
package com.wangzhen.adapter.sample.holder;

import android.view.View;

import com.wangzhen.adapter.base.RecyclerItem;
import com.wangzhen.adapter.sample.MainActivity;
import com.wangzhen.adapter.sample.R;

/**
 * EmptyHolder
 * Created by wangzhen on 2020/6/10.
 */
public class EmptyHolder extends RecyclerItem {
    @Override
    public int layout() {
        return R.layout.item_empty_layout;
    }

    @Override
    public void onViewCreated(View itemView) {
        itemView.findViewById(R.id.btn_retry).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (v.getContext() instanceof MainActivity) {
                    ((MainActivity) v.getContext()).init();
                }
            }
        });
    }
}

File: app/src/main/java/com/wangzhen/adapter/sample/holder/EmptyHolder.kt
package com.wangzhen.adapter.sample.holder

import android.view.View
import com.wangzhen.adapter.base.RecyclerItem
import com.wangzhen.adapter.sample.LoadMoreActivity
import com.wangzhen.adapter.sample.R

/**
 * EmptyHolder
 * Created by wangzhen on 2021/5/25
 */
class EmptyHolder : RecyclerItem() {
    public override fun layout(): Int {
        return R.layout.item_empty_layout
    }

    override fun onViewCreated(itemView: View) {
        itemView.findViewById<View>(R.id.btn_retry).setOnClickListener { v ->
            if (v.context is LoadMoreActivity) {
                (v.context as LoadMoreActivity).bind()
            }
        }
    }
}